libbeat/autodiscover/autodiscover_test.go

```diff
@@ -186,3 +187,60 @@ func TestAutodiscover(t *testing.T) {
 	assert.True(t, runners[0].started)
 	assert.True(t, runners[0].stopped)
 }
+
+func TestAutodiscoverHash(t *testing.T) {
+	// Register mock autodiscover provider
+	busChan := make(chan bus.Bus, 1)
+
+	ProviderRegistry = NewRegistry()
+	ProviderRegistry.AddProvider("mock", func(b bus.Bus, c *common.Config) (Provider, error) {
+		// intercept bus to mock events
+		busChan <- b
+
+		return &mockProvider{}, nil
+	})
+
+	// Create a mock adapter
+	runnerConfig1, _ := common.NewConfigFrom(map[string]string{
+		"runner": "1",
+	})
+	runnerConfig2, _ := common.NewConfigFrom(map[string]string{
+		"runner": "2",
+	})
+	adapter := mockAdapter{
+		configs: []*common.Config{runnerConfig1, runnerConfig2},
+	}
+
+	// and settings:
+	providerConfig, _ := common.NewConfigFrom(map[string]string{
+		"type": "mock",
+	})
+	config := Config{
+		Providers: []*common.Config{providerConfig},
+	}
+
+	// Create autodiscover manager
+	autodiscover, err := NewAutodiscover("test", &adapter, &config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Start it
+	autodiscover.Start()
+	defer autodiscover.Stop()
+	eventBus := <-busChan
+
+	// Test start event
+	eventBus.Publish(bus.Event{
+		"start": true,
+		"meta": common.MapStr{
+			"foo": "bar",
+		},
+	})
+	time.Sleep(10 * time.Millisecond)
+	runners := adapter.Runners()
+	assert.Equal(t, len(runners), 2)
```

Comment:
I don't fully understand what's being tested here so I'll just ask if it it needs to be asserting anything on the second runner.

Reply:
The main thing tested here is that 2 runner are running. Without the previous changes, only 1 runner was running. I will add the checks below also for runner 2 to be sure.